File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserCheerAnimalRes.java
package com.team05.petmeeting.domain.user.dto.profile;

import com.team05.petmeeting.domain.animal.entity.Animal;

import java.util.List;

public record UserCheerAnimalRes(
        long totalAnimalCount,
        List<CheerAnimalItem> animals
) {
    public static UserCheerAnimalRes from(List<Object[]> animalCountMap) {
        List<CheerAnimalItem> items = animalCountMap.stream()
                .map(row -> CheerAnimalItem.from((Animal) row[0], (Long) row[1]))
                .toList();

        return new UserCheerAnimalRes(items.size(), items);
    }

    public record CheerAnimalItem(
            Long animalId,
            String species,      // upKindNm (개/고양이)
            String breed,        // kindFullNm (믹스견 등)
            String imageUrl,     // popfile1
            long myCheerCount,   // "5개 보냄" (내가 이 동물에게 보낸 개수)
            double temperature   // "85.0C" (동물의 총 응원 수 기반 온도)
    ) {
        public static CheerAnimalItem from(Animal animal, long myCheerCount) {
            return new CheerAnimalItem(
                    animal.getId(),
                    animal.getUpKindNm(),
                    animal.getKindFullNm(),
                    animal.getPopfile1(),
                    myCheerCount,
                    animal.getTemperature()
            );
        }
    }

}

File: backend/src/main/java/com/team05/petmeeting/domain/user/dto/profile/UserCheerAnimalRes.kt
package com.team05.petmeeting.domain.user.dto.profile

import com.team05.petmeeting.domain.animal.entity.Animal

data class UserCheerAnimalRes(
    val totalAnimalCount: Long,
    val animals: MutableList<CheerAnimalItem>
) {

    data class CheerAnimalItem(
        val animalId: Long,
        val species: String,  // upKindNm (개/고양이)
        val breed: String,  // kindFullNm (믹스견 등)
        val imageUrl: String,  // popfile1
        val myCheerCount: Long,  // "5개 보냄" (내가 이 동물에게 보낸 개수)
        val temperature: Double // "85.0C" (동물의 총 응원 수 기반 온도)
    ) {
        companion object {
            @JvmStatic
            fun from(animal: Animal, myCheerCount: Long): CheerAnimalItem {
                return CheerAnimalItem(
                    animal.id,
                    animal.upKindNm,
                    animal.kindFullNm,
                    animal.popfile1,
                    myCheerCount,
                    animal.getTemperature()
                )
            }
        }
    }

    companion object {
        @JvmStatic
        fun from(animalCountMap: MutableList<Array<Any>>): UserCheerAnimalRes {
            val items = animalCountMap.stream()
                .map { row: Array<Any> ->
                    CheerAnimalItem.from(
                        (row[0] as Animal),
                        (row[1] as Long)
                    )
                }
                .toList()

            return UserCheerAnimalRes(items.size.toLong(), items)
        }
    }
}
